### DISCUSSION TITLE
Using recycle option to save money

### DISCUSSION THREAD

**Discussion by jacobg** [diegoazh/gmap-vue#312] [Q&A]
It looks like if you set the `options` prop on `GmapMap`  to `{ recycle: 'SOME_KEY'` }, then the component will re-use the same Google Map on every instance of the Vue component in order to save money on map loads. Is that the correct understanding?

**Comment by diegoazh** [comment]
In general yes, but we use a default prefix on the `__gmc__`, and if you leave the recycle key empty it is used as a fallback. That was the [issue reported](#309).

After discovering that, we decided to use a `Map` data structure to save every promise of every component and to have control over any of them. These changes give us more control over the instances, to destroy them when the component is unmounted, and to get the correct instance when using a composable.

For example, before this change, if you have two maps and use the composable `useMapPromise()` to get the promise of your map we can't know which map you are requesting. With markers, this is worse because you can have multiple markers and when you use the `useMarkerPromise()` we don't know which marker you are requesting.

Now saving the promises on a `Map` data structure every promise is registered with the provided key and you can request a specific instance using `useMarkerPromise('youMarkerKey')`.

I hope I have been able to explain myself.
